src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1129,12 +1130,12 @@ static BluezConnection * BluezCharacteristicGetBluezConnection(BluezGattCharacte
         g_variant_dict_init(&options, aOptions);
 
         v = g_variant_dict_lookup_value(&options, "device", G_VARIANT_TYPE_OBJECT_PATH);
+        VerifyOrExit(v != nullptr, ChipLogError(DeviceLayer, "FAIL: No device option in dictionary (%s)", __func__));
```

Comment:
Should options be cleared on failure as well?

Reply:
Fixed